Signal/src/ViewControllers/HomeView/HomeViewCell.m

```diff
@@ -0,0 +1,417 @@
+@property (nonatomic) UILabel *unreadLabel;
+
+@property (nonatomic, nullable) TSThread *thread;
+@property (nonatomic, nullable) OWSContactsManager *contactsManager;
+
+@property (nonatomic, readonly) NSMutableArray<NSLayoutConstraint *> *viewConstraints;
+
+@end
+
+#pragma mark -
+
+@implementation HomeViewCell
+
+- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(nullable NSString *)reuseIdentifier
+{
+    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
+        [self commontInit];
+    }
+    return self;
+}
+
+// `[UIView init]` invokes `[self initWithFrame:...]`.
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    if (self = [super initWithFrame:frame]) {
+        [self commontInit];
+    }
+
+    return self;
+}
+
+- (void)commontInit
+{
+    OWSAssert(!self.avatarView);
+
+    [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.layoutMargins = UIEdgeInsetsZero;
+    self.contentView.layoutMargins = UIEdgeInsetsZero;
+    self.preservesSuperviewLayoutMargins = NO;
+    self.contentView.preservesSuperviewLayoutMargins = NO;
+
+    self.backgroundColor = [UIColor whiteColor];
+
+    _viewConstraints = [NSMutableArray new];
+
+    self.avatarView = [[AvatarImageView alloc] init];
+    [self.contentView addSubview:self.avatarView];
+    [self.avatarView autoSetDimension:ALDimensionWidth toSize:kHomeViewAvatarSize];
+    [self.avatarView autoSetDimension:ALDimensionHeight toSize:kHomeViewAvatarSize];
+    [self.avatarView autoPinLeadingToSuperviewMarginWithInset:kHomeViewCellHMargin];
+    [self.avatarView autoVCenterInSuperview];
+    [self.avatarView setContentHuggingHigh];
+    [self.avatarView setCompressionResistanceHigh];
+
+    self.payloadView = [UIView containerView];
+    [self.contentView addSubview:self.payloadView];
+    [self.payloadView autoPinLeadingToTrailingEdgeOfView:self.avatarView offset:kHomeViewAvatarHSpacing];
+    [self.payloadView autoVCenterInSuperview];
+
+    self.nameLabel = [UILabel new];
+    self.nameLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+    self.nameLabel.font = self.nameFont;
+    [self.nameLabel setContentHuggingHorizontalLow];
+    [self.nameLabel setCompressionResistanceHorizontalLow];
+
+    self.dateTimeLabel = [UILabel new];
+    [self.dateTimeLabel setContentHuggingHorizontalHigh];
+    [self.dateTimeLabel setCompressionResistanceHorizontalHigh];
+
+    UIStackView *topRowView = [[UIStackView alloc] initWithArrangedSubviews:@[
+        self.nameLabel,
+        self.dateTimeLabel,
+    ]];
+    topRowView.axis = UILayoutConstraintAxisHorizontal;
+    topRowView.spacing = 4;
+    [self.payloadView addSubview:topRowView];
+    [topRowView autoPinLeadingToSuperviewMargin];
+    [topRowView autoPinTrailingToSuperviewMargin];
+    [topRowView autoPinTopToSuperviewMargin];
+
+    self.snippetLabel = [UILabel new];
+    self.snippetLabel.font = [self snippetFont];
+    self.snippetLabel.numberOfLines = 1;
```

Comment:
Yup, that's right. It was part of the rationale for doing away with the dividers, since the snippet text will now always be one line they can carry some of the design burden of defining the "edge" of a cell.